### ISSUE TITLE
`doc-valid-idents` not respected in Cargo.toml [lints] table

### ISSUE THREAD

**Issue by rockboynton** [rust-lang/rust-clippy#11751]
### Summary

See title

### Reproducer

I tried this code:

```toml
[lints.clippy]
pedantic    = "deny"
doc_markdown = { level = "deny", doc_valid_idents = ["GHz", ".."] }
```
```rust
//! GHz
```

I expected to see this happen:

A clean build

Instead, this happened:

I still see the error on the `GHz` in my doc comment

### Version

```text
❯ rustc -Vv
rustc 1.74.0-nightly (58eefc33a 2023-08-24)
binary: rustc
commit-hash: 58eefc33adf769a1abe12ad94b3e6811185b4ce5
commit-date: 2023-08-24
host: x86_64-unknown-linux-gnu
release: 1.74.0-nightly
LLVM version: 17.0.0
```

**Comment by Stoned33** [comment]
Lints in Rust can be controlled and customized using the `#[allow(lint_name)]` attribute in the code. This attribute allows you to selectively disable specific lints for specific sections of code. While this attribute is typically used within the code itself, it may be worth exploring if it can be applied to the Cargo.toml file as well.

**Comment by rockboynton** [comment]
Hi @Stoned33, thanks for your reply. While that does work per crate, other lints work with overriding them in the Cargo.toml as in my example, although I do notice it only seems to work when it's only the bare level key.

For example, this works as expected, to allow denying the pedantic group with an exception, since the order matters in the file

```toml
[lints.clippy]
pedantic    = "deny"
borrow_as_ptr = "allow" 
```

**Comment by samueltardieu** [comment]
Isn't lint extra configuration supposed to be given in `clippy.toml`? If you create one with

```toml
doc-valid-idents = ["GHz", ".."]
```

it should do what you want.

**Comment by Alexendoo** [comment]
Opened a cargo issue for this: https://github.com/rust-lang/cargo/issues/12917
